AsyncDisplayKit/Details/_ASDisplayLayer.mm

```diff
@@ -122,6 +122,12 @@ - (void)setNeedsLayout
   ASDisplayNodeAssertMainThread();
   [super setNeedsLayout];
 }
+
+- (void)layoutIfNeeded
+{
+  ASDisplayNodeAssertMainThread();
+  [super layoutIfNeeded];
```

Comment:
I made `layoutIfNeeded` as well as `setNeedsLayout` safe on `_ASDisplayLayer.m` and `_ASDisplayView.m`. They both trampoline to the main thread now.